packages/contracts/contract/src/actions/transactions.rs

```diff
@@ -1,28 +1,171 @@
-use near_sdk::{env, near_bindgen, AccountId, Promise};
+use near_sdk::{env, near_bindgen, AccountId, Promise, json_types::U128, is_promise_success};
 use near_bigint::U256;
 use near_plonk_verifier::{Proof, G1Point};
 
+use crate::*;
 use crate::{
-  Contract, ContractExt,
   hashes::{account_hash, serial_hash},
   events::{event_deposit, event_withdrawal},
 };
 
+const CALLBACK_GAS: Gas = Gas(30_000_000_000_000);
+
+#[ext_contract(ext_self)]
+pub trait ExtSelf {
+  fn withdraw_callback(
+    nullifier_hash: U256,
+    recipient: AccountId,
+    relayer: Option<AccountId>,
+    fee: U256,
+    refund: U256,
+  );
+}
+
 #[near_bindgen]
 impl Contract {
   #[payable]
   pub fn deposit(&mut self, secrets_hash: U256) {
+    assert!(
```

Comment:
I'd suggest to wrap this assertion in a function, the same way you did with the owner

Reply:
Why would I wrap it around a function if this is the only place where it is used? Is there any other place that I'm missing? Couldn't find any using ctrl + shift + F on VSCode